usl_pipeline/study_area_uploader/study_area_uploader/chunkers/study_area_chunkers.py

```diff
@@ -0,0 +1,160 @@
+import io
+import logging
+import pathlib
+import tarfile
+from typing import Optional, Tuple
+
+from google.cloud import storage
+from shapely import geometry
+
+from study_area_uploader.chunkers import elevation_chunkers
+from study_area_uploader.transformers import study_area_transformers
+from usl_lib.chunkers import polygon_chunkers
+from usl_lib.readers import elevation_readers
+from usl_lib.shared import geo_data
+from usl_lib.storage import file_names
+from usl_lib.writers import elevation_writers
+
+
+def _chunk_polygons_if_present(
+    elevation_header: geo_data.ElevationHeader,
+    chunk_size: int,
+    polygon_masks: Optional[list[Tuple[geometry.Polygon, int]]],
+    work_dir: pathlib.Path,
+    object_type_name: str,
+    support_mask_values: bool = False,
+    chunk_additional_border_cells: int = 0,
+) -> Optional[pathlib.Path]:
+    """Splits optional list of polygons with masks (if present) into chunk files."""
+    logging.info("Preparing %s chunks...", object_type_name)
+    chunks_dir = None
+    if polygon_masks is not None:
+        chunks_dir = work_dir / f"{object_type_name}_chunks"
+        chunks_dir.mkdir(parents=True, exist_ok=True)
+        polygon_chunkers.split_polygons_into_chunks(
+            elevation_header,
+            chunk_size,
+            polygon_masks,
+            chunks_dir,
+            support_mask_values=support_mask_values,
+            chunk_additional_border_cells=chunk_additional_border_cells,
+        )
+    return chunks_dir
+
+
+def _add_chunk_to_tar_if_present(
+    chunks_dir: Optional[pathlib.Path],
+    chunk_file_name: str,
+    tar_entry_file_name: str,
+    tar: tarfile.TarFile,
+):
+    """Copy new entry to TAR-file from a chunk file if the folder with chunks exists."""
+    if chunks_dir is not None:
+        tar.add(str(chunks_dir / chunk_file_name), arcname=tar_entry_file_name)
+
+
+def build_and_upload_chunks(
+    study_area_name: str,
+    input_data: study_area_transformers.PreparedInputData,
+    work_dir: pathlib.Path,
+    study_area_chunk_bucket: storage.Bucket,
+    chunk_size: int = 1000,
```

Comment:
Since I actually do have default value = 1000 in the uploader main.py CLI args (https://github.com/UrbanSystemsLab/climateiq-cnn/blob/main/usl_pipeline/study_area_uploader/study_area_uploader/main.py#L145) I don't need to have it hardcoded in the code. On the cloud_function side though I would prefer to load it from metadata instead of having it hardcoded...

Reply:
Retrieving the chunk size from metadata in the cloud function makes sense to me. Do you have a plan for how to get that into the metastore?

Reply:
I guess we can add some new field called `chunk_size` and update it during the chunking stage?

Reply:
yep, sounds good
that does require all chunks being the same size (or else each chunk would attempt to set a different chunk_size for the study area if it's assuming it's own size is the correct size) which lines up with what we'd decided previously.